registration-processor/core-processor/registration-processor-reprocessor-stage/src/main/java/io/mosip/registration/processor/reprocessor/service/WorkflowSearchService.java

```diff
@@ -0,0 +1,137 @@
+package io.mosip.registration.processor.reprocessor.service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.stereotype.Component;
+
+import io.mosip.kernel.core.dataaccess.exception.DataAccessLayerException;
+import io.mosip.kernel.core.exception.ExceptionUtils;
+import io.mosip.kernel.core.logger.spi.Logger;
+import io.mosip.registration.processor.core.code.ModuleName;
+import io.mosip.registration.processor.core.constant.LoggerFileConstant;
+import io.mosip.registration.processor.core.exception.WorkFlowSearchException;
+import io.mosip.registration.processor.core.exception.WorkflowActionException;
+import io.mosip.registration.processor.core.exception.util.PlatformErrorMessages;
+import io.mosip.registration.processor.core.exception.util.PlatformSuccessMessages;
+import io.mosip.registration.processor.core.logger.LogDescription;
+import io.mosip.registration.processor.core.logger.RegProcessorLogger;
+import io.mosip.registration.processor.core.workflow.dto.PageResponseDto;
+import io.mosip.registration.processor.core.workflow.dto.SearchDto;
+import io.mosip.registration.processor.core.workflow.dto.SearchResponseDto;
+import io.mosip.registration.processor.rest.client.audit.builder.AuditLogRequestBuilder;
+import io.mosip.registration.processor.status.dto.InternalRegistrationStatusDto;
+import io.mosip.registration.processor.status.dto.RegistrationStatusDto;
+import io.mosip.registration.processor.status.service.RegistrationStatusService;
+
+@Component
+public class WorkflowSearchService {
+
+	/** The registration status service. */
+	@Autowired
+	RegistrationStatusService<String, InternalRegistrationStatusDto, RegistrationStatusDto> registrationStatusService;
+
+	/** The core audit request builder. */
+	@Autowired
+	AuditLogRequestBuilder auditLogRequestBuilder;
+
+	/** The module name. */
+	public static String MODULE_NAME = ModuleName.WORKFLOW_ACTION_SERVICE.toString();
+
+	/** The module id. */
+	public static String MODULE_ID = PlatformSuccessMessages.RPR_WORKFLOW_ACTION_SERVICE_SUCCESS.getCode();
+
+	/** The reg proc logger. */
+	private static Logger regProcLogger = RegProcessorLogger.getLogger(WorkflowActionService.class);
+
+	public PageResponseDto<SearchResponseDto> searchRegistrationDetails(SearchDto searchDto)
+			throws WorkFlowSearchException {
+
+		PageResponseDto<SearchResponseDto> pageResponseDto = new PageResponseDto<>();
+		List<SearchResponseDto> searchResDtos = new ArrayList<SearchResponseDto>();
+		regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.USERID.toString(), "",
+				"WorkflowSearchService::searchRegistrationDetails()::entry");
+		try {
+			Page<InternalRegistrationStatusDto> pageDtos = registrationStatusService.searchRegistrationDetails(searchDto);
```

Comment:
Here validate the field names in the searchInfo object and convert to the actual column names, like workflowId to registrationId and workflowType to registrationType etc..
On validation result throw a specific exception and let the API handle and return back the specific error response

Reply:
made changes related to workflowId in the request and taken care workflowId to convert to id in dao

Reply:
As per dao it should not understand what is workflow Id, that conversion should be understood only by the WorkflowSearchService, also what about the validation of field in filter and sort. what if someone sends a field as part of filter in the request that is not available in the table ?

Reply:
we r doing validation for filter sort and pagination

Reply:
taken care